java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -929,74 +929,127 @@ public ColumnBuilder appendStructValues(StructData... inputList) {
     }
 
     /**
-     * A method that is responsible for growing the buffers as needed
-     * and incrementing the row counts when we append values or nulls.
-     * @param hasNull indicates whether the validity buffer needs to be considered, as the
-     *                nullcount may not have been fully calculated yet
-     * @param length used for strings
+     * Grows valid buffer lazily. The valid buffer won't be materialized until the first null
+     * value appended. This method reuses the rowCapacity to track the sizes of column.
+     * Therefore, please call specific growBuffer method to update rowCapacity before calling
+     * this method.
+     */
+    private void growValidBuffer() {
+      long maskBytes = byteSizeOfNullMask((int) rowCapacity);
```

Comment:
Would it be better to have a `validCapacity` in addition to `rowCapacity`? So we only have to recalculate this when we actually grow the validity mask?

Reply:
Yes, I introduced an independent variable to record `validCapacity`.